airbyte/_executors/util.py

```diff
@@ -143,49 +195,23 @@ def get_connector_executor(  # noqa: PLR0912, PLR0913, PLR0915 # Too complex
         )
 
     if source_manifest:
-        if source_manifest is True:
-            # Auto-set the manifest to a valid http address URL string
-            source_manifest = (
-                "https://raw.githubusercontent.com/airbytehq/airbyte/master/airbyte-integrations"
-                f"/connectors/{name}/{name.replace('-', '_')}/manifest.yaml"
+        if isinstance(source_manifest, (dict, Path)):
+            return DeclarativeExecutor(
+                name=name,
+                manifest=source_manifest,
             )
-        if isinstance(source_manifest, str):
-            print("Installing connector from YAML manifest:", source_manifest)
-            # Download the manifest file
-            response = requests.get(url=source_manifest)
-            response.raise_for_status()  # Raise an exception if the download failed
-
-            if "class_name:" in response.text:
-                raise exc.AirbyteConnectorInstallationError(
-                    message=(
-                        "The provided manifest requires additional code files (`class_name` key "
-                        "detected). This feature is not compatible with the declarative YAML "
-                        "executor. To use this executor, please try again with the Python "
-                        "executor."
-                    ),
-                    connector_name=name,
-                    context={
-                        "manifest_url": source_manifest,
-                    },
-                )
 
-            try:
-                source_manifest = cast(dict, yaml.safe_load(response.text))
-            except JSONDecodeError as ex:
-                raise exc.AirbyteConnectorInstallationError(
-                    connector_name=name,
-                    context={
-                        "manifest_url": source_manifest,
-                    },
-                ) from ex
-
-        if isinstance(source_manifest, Path):
-            source_manifest = cast(dict, yaml.safe_load(source_manifest.read_text()))
-
-        # Source manifest is a dict at this point
-        return DeclarativeExecutor(
-            manifest=source_manifest,
-        )
+        if isinstance(source_manifest, (str, bool)):
+            # Source manifest is either a URL or a boolean (True)
+            source_manifest = _try_get_source_manifest(
+                source_name=name,
+                manifest_url=None if source_manifest is True else source_manifest,
```

Comment:
I'm trying to figure out a bit more about PyAirbyte ; in which case `get_connector_executor` is called with `source_manifest` being a URL?